presto-orc/src/main/java/com/facebook/presto/orc/StripeReader.java

```diff
@@ -475,7 +510,15 @@ public StripeFooter readStripeFooter(StripeId stripeId, StripeInformation stripe
         int footerLength = toIntExact(stripe.getFooterLength());
 
         // read the footer
-        Slice footerSlice = stripeMetadataSource.getStripeFooterSlice(orcDataSource, stripeId, footerOffset, footerLength, cacheable);
+        Slice footerSlice = null;
+        if (stripeMetaCache.isPresent()) {
```

Comment:
This is where we get the stripe footer from the stripe metadata cache. We would then proceed to read the stripe footer directly from the source if it was not in the cache.